kobo-uncaged/device/types.go

```diff
@@ -15,30 +15,54 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with Kobo UNCaGED.  If not, see <https://www.gnu.org/licenses/>.
 
-package main
+package device
 
 import (
+	"database/sql"
 	"fmt"
 	"image"
 	"path/filepath"
+	"regexp"
 	"strings"
+	"sync"
 
 	"github.com/bamiaux/rez"
+	"github.com/shermp/Kobo-UNCaGED/kobo-uncaged/util"
+	"github.com/shermp/UNCaGED/uc"
 )
 
 type cidPrefix string
 
-type mboxSection int
-type kuPrinter interface {
-	kuPrintln(section mboxSection, a ...interface{}) (n int, err error)
-	kuClose()
+// KuOptions contains some options that are required
+type KuOptions struct {
+	PreferSDCard bool
+	PreferKepub  bool
+	PasswordList []string
+	Thumbnail    thumbnailOption
 }
 
-const (
-	header mboxSection = iota
-	body
-	footer
-)
+// Kobo contains the variables and methods required to use
+// the UNCaGED library
+type Kobo struct {
+	Device            koboDevice
+	fw                [3]int
+	KuConfig          *KuOptions
+	DBRootDir         string
+	BKRootDir         string
+	ContentIDprefix   cidPrefix
+	useSDCard         bool
+	MetadataMap       map[string]KoboMetadata
+	UpdatedMetadata   []string
+	Passwords         *uncagedPassword
+	DriveInfo         uc.DeviceInfo
+	nickelDB          *sql.DB
+	Wg                *sync.WaitGroup
+	InvalidCharsRegex *regexp.Regexp
```

Comment:
I would make this a global.

Reply:
I moved the filepath sanitation (and regex) to the `util` package. Seemed a better fit there.